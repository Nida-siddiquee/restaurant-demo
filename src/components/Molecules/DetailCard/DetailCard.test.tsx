import { render, screen, fireEvent } from '@testing-library/react';
import RestaurantCard from './DetailCard';

describe('RestaurantCard', () => {
  const baseProps = {
    name: 'Testaurant',
    rating: 4.7,
    reviewCount: '100',
    deliveryTime: '20-30 min',
    deliveryFee: '£2.00',
    onClick: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders all core info', () => {
    render(<RestaurantCard {...baseProps} />);
    expect(screen.getByText('Testaurant')).toBeInTheDocument();
    expect(screen.getByText('★')).toBeInTheDocument();
    expect(screen.getByText('(100)')).toBeInTheDocument();
    expect(screen.getByText(/20-30 min/)).toBeInTheDocument();
    expect(screen.getByText(/Delivery from £2.00/)).toBeInTheDocument();
  });

  it('renders the logo image if logoUrl is provided', () => {
    render(<RestaurantCard {...baseProps} logoUrl="logo.png" />);
    const logo = screen.getByAltText('Testaurant logo');
    expect(logo).toBeInTheDocument();
    expect(logo).toHaveAttribute('src', 'logo.png');
  });

  it('renders the hero image if heroUrl is provided', () => {
    render(<RestaurantCard {...baseProps} heroUrl="hero.png" />);
    const hero = screen.getByAltText('Testaurant banner');
    expect(hero).toBeInTheDocument();
    expect(hero).toHaveAttribute('src', 'hero.png');
  });

  it('renders the offer ribbon if offer is provided', () => {
    render(<RestaurantCard {...baseProps} offer="20% Off!" />);
    expect(screen.getByText(/20% Off!/)).toBeInTheDocument();
    // Optionally: check the tag icon by role or svg tag
    expect(screen.getByText(/20% Off!/).parentElement?.querySelector('svg')).toBeInTheDocument();
  });

  it('renders the badge if badge is provided', () => {
    render(<RestaurantCard {...baseProps} badge="Sponsored" />);
    expect(screen.getByText('Sponsored')).toBeInTheDocument();
  });

  it('calls onClick when card is clicked', () => {
    render(<RestaurantCard {...baseProps} />);
    fireEvent.click(screen.getByText('Testaurant'));
    expect(baseProps.onClick).toHaveBeenCalled();
  });

  it('highlights the name if highlight is provided', () => {
    render(<RestaurantCard {...baseProps} highlight="Test" />);
    expect(screen.getByText('Test')).toContainHTML('mark');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <RestaurantCard
        {...baseProps}
        logoUrl="logo.png"
        heroUrl="hero.png"
        offer="Deal"
        badge="StampCard"
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
