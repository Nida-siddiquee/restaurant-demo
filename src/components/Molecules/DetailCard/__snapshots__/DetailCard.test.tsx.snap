// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background: white;
  border-radius: 12px;
  box-shadow: 0 4px 16px rgba(25, 40, 70, 0.06);
  overflow: hidden;
  margin: 0.7rem 0;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  cursor: pointer;
  transition: box-shadow 0.2s;
}

.c0:hover {
  box-shadow: 0 8px 28px rgba(25, 40, 70, 0.13);
}

.c1 {
  position: relative;
  width: 100%;
  height: 150px;
  overflow: hidden;
}

.c1 img.banner {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c2 {
  position: absolute;
  top: 10px;
  left: 10px;
  background: #fff4d9;
  color: #6a4600;
  font-weight: 600;
  font-size: 0.86rem;
  padding: 4px 10px;
  border-radius: 40px;
  display: flex;
  align-items: center;
  gap: 0.35rem;
  box-shadow: 0 2px 6px rgba(0, 0, 0, 0.1);
}

.c2 svg {
  flex-shrink: 0;
}

.c4 {
  position: absolute;
  bottom: 12px;
  right: 12px;
  background: #111;
  color: #fff;
  font-size: 0.75rem;
  font-weight: 600;
  padding: 3px 8px;
  border-radius: 4px;
  opacity: 0.9;
}

.c3 {
  position: absolute;
  bottom: 10px;
  left: 12px;
  width: 70px;
  height: 70px;
  object-fit: contain;
  background: #fff;
  border-radius: 12px;
  border: 2px solid #eee;
}

.c5 {
  padding: 1.1rem 1.1rem 0.9rem;
  flex: 1;
  display: flex;
  flex-direction: column;
}

.c6 {
  font-size: 1.25rem;
  font-weight: 700;
  margin: 0 0 0.45rem;
}

.c7 {
  display: flex;
  align-items: center;
  color: #606770;
  font-size: 1rem;
  margin-bottom: 0.4rem;
  gap: 0.5em;
}

.c7 .star {
  color: #f5a623;
}

.c7 .clock {
  font-size: 1em;
}

.c8 {
  display: flex;
  align-items: center;
  gap: 0.5em;
  font-size: 1.05rem;
  font-weight: 500;
  color: #222;
}

@media (max-width: 600px) {
  .c0 {
    margin: 0.6rem 0 1rem 0;
    border-radius: 9px;
  }
}

@media (max-width: 600px) {
  .c1 {
    height: 90px;
  }
}

@media (max-width: 600px) {
  .c2 {
    top: 6px;
    left: 6px;
    font-size: 0.76rem;
    padding: 3px 8px;
  }
}

@media (max-width: 600px) {
  .c4 {
    bottom: 6px;
    right: 6px;
    font-size: 0.67rem;
    padding: 2px 7px;
  }
}

@media (max-width: 600px) {
  .c3 {
    width: 42px;
    height: 42px;
    bottom: 6px;
    left: 6px;
    border-radius: 7px;
    border-width: 1.2px;
  }
}

@media (max-width: 600px) {
  .c5 {
    padding: 0.7rem 0.7rem 0.55rem;
  }
}

@media (max-width: 600px) {
  .c6 {
    font-size: 1rem;
    margin-bottom: 0.28rem;
  }
}

@media (max-width: 600px) {
  .c7 {
    font-size: 0.88rem;
    gap: 0.33em;
    margin-bottom: 0.22rem;
  }
}

@media (max-width: 600px) {
  .c8 {
    font-size: 0.9rem;
    gap: 0.32em;
  }
}

<div
    aria-label="Testaurant restaurant - 4.7 stars, 20-30 min delivery, from ¬£2.00, Deal"
    class="c0"
    role="button"
    tabindex="0"
  >
    <div
      class="c1"
    >
      <img
        alt="Testaurant banner"
        class="banner"
        src="hero.png"
      />
      <div
        class="c2"
      >
        <svg
          fill="#ff8000"
          height="14"
          viewBox="0 0 24 24"
          width="14"
        >
          <path
            d="M21.586 11.586l-9.172-9.172a2.002 2.002 0 0 0-1.414-.586h-7a2 2 0 0 0-2 2v7c0 .53.211 1.039.586 1.414l9.172 9.172a2 2 0 0 0 2.828 0l7-7a2 2 0 0 0 0-2.828zM7.5 8a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
          />
        </svg>
        Deal
      </div>
      <img
        alt="Testaurant logo"
        class="c3"
        src="logo.png"
      />
      <div
        class="c4"
      >
        StampCard
      </div>
    </div>
    <div
      class="c5"
    >
      <h3
        class="c6"
      >
        Testaurant
      </h3>
      <div
        class="c7"
      >
        <span
          aria-hidden="true"
          class="star"
        >
          ‚òÖ
        </span>
        <span
          aria-label="Rating: 4.7 out of 5 stars"
        >
          4.7
        </span>
        <span
          aria-label="100 reviews"
          style="font-size: .9em; color: rgb(136, 102, 136);"
        >
          (100)
        </span>
        <span
          aria-hidden="true"
          class="clock"
        >
          üïí
        </span>
        <span
          aria-label="Delivery time: 20-30 min"
        >
          20-30 min
        </span>
      </div>
      <div
        class="c8"
      >
        <span
          aria-hidden="true"
        >
          üö¥‚Äç‚ôÇÔ∏è
        </span>
        <span
          aria-label="Delivery cost: ¬£2.00"
        >
          Delivery from ¬£2.00
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
